explorer/server.go

```diff
@@ -122,47 +122,66 @@ func (a *App) getNode(w http.ResponseWriter, r *http.Request) {
 		log.Warn().Str("Couldn't find entry to redis", string(err.Error())).Msg("")
 
 	}
-	if value != "" {
-		w.Write([]byte(value))
-		return
+	// No value, fetch data from the node
+	if value == "" {
+		nodeInfo, err := a.fetchNodeData(r.Context(), nodeID)
+		if err != nil {
+			log.Error().Err(errors.Wrap(err, "could not fetch node data")).Msg("")
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte(http.StatusText(http.StatusInternalServerError)))
+			return
+		}
+		// Save value in redis
+		// caching for 30 mins
+		marshalledInfo, err := json.Marshal(nodeInfo)
+		if err != nil {
+			log.Error().Err(errors.Wrap(err, "could not marshal node info")).Msg("")
```

Comment:
I think u should return an error to the user here, and terminate the execution of this handler!

Reply:
Also why the weird call ? this should be like this
```suggestion
			log.Error().Err(err).Msg("could not marshal node info")
```